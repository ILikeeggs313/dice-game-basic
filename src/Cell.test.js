import React from "react";
import { render, asFragment } from "@testing-library/react";
import Cell from "./Cell";

//describe the general test here
describe("Testing rendering of the cell component", () => {
    let container;
    //let's add a beforeEach
    beforeEach( ()=> {
        let tr = document.createElement("tr");
        container = document.body.appendChild(tr);
    });



    //a basic smoke test to see if it renders without crashing
    test("renders without crashing", () => {
        render(< Cell />, {container})
    })

    //test if it matches snapshot when the board is lit up
    test("matches snapshot as lit up", () => {
        const {asFragment} = render(<Cell isLit/>, {container});
        expect(asFragment()).toMatchSnapshot()
    })

    test("matches snapshot when not lit", () => {
        const {asFragment} = render(<Cell  />, {container})
        expect(asFragment()).toMatchSnapshot()
    })

})