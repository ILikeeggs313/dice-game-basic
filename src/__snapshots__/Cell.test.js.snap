// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing rendering of the cell component matches snapshot as lit up 1`] = `<DocumentFragment />`;

exports[`Testing rendering of the cell component matches snapshot when not lit 1`] = `<DocumentFragment />`;
